python/pyarrow/tests/test_flight.py

```diff
@@ -2539,36 +2565,60 @@ def received_headers(self, headers):
 
 
 def test_flight_dictionary_deltas_do_exchange():
+    expected_dict_deltas_stats = ReadStats(
```

Comment:
Small suggestion: if you want to avoid the `if`s below, you can make this a dict:
```python
    expected_stats = {
        b'dict_replacement': ...
```

Reply:
Thanks for the suggestion, I've pushed the test improvement and I've rebased main to fix the LLVM failures on CI